Firestore/core/src/firebase/firestore/model/field_path.cc

```diff
@@ -59,6 +59,23 @@ bool IsValidIdentifier(const std::string& segment) {
   return true;
 }
 
+/** A custom formatter to be used with absl::StrJoin(). */
+struct JoinEscaped {
```

Comment:
Have you considered using `DISABLE_GOOGLE_STRING` macro instead?

Reply:
It is a good ideal for building a particular app. However, may not work for library. Say a 1p app depends on Firestore and Foo, where we disable_google_string while Foo does not and both depends on Bar, where DISABLE_GOOGLE_STRING matters ... The behavior is not obviously when Bar is involved.